runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5278,6 +5287,40 @@ class INTERPRETER_CLASS
 			} else if (J9NtcPointer == argType) {
 				/* ffi_call expects the address of the pointer is the address of the stackslot. */
 				pointerValues[i] = (U_64)ffiArgs[i];
+#if JAVA_SPEC_VERSION >= 22
+				if (J9_ARE_ALL_BITS_SET(pointerValues[i], J9_FFI_DOWNCALL_HEAP_ARGUMENT_ID)) {
+					U_64 heapOffset = *(U_64 *)J9JAVAARRAY_EA(_currentThread,
+							J9_JNI_UNWRAP_REFERENCE(_sp + 10), /* The heap offset array */
+							curHeapArgIdx,
+							U_64);
+					j9object_t heapBase = (j9object_t)J9JAVAARRAYOFOBJECT_LOAD(_currentThread,
+							J9_JNI_UNWRAP_REFERENCE(_sp + 11), /* The heap base array */
+							curHeapArgIdx);
```

Comment:
Please put the first argument on a line by itself like the others, and add the missing period:
```c
					j9object_t heapBase = (j9object_t)J9JAVAARRAYOFOBJECT_LOAD(
							_currentThread,
							J9_JNI_UNWRAP_REFERENCE(_sp + 11), /* The heap base array. */
							curHeapArgIdx);
```

Reply:
Fixed.